sw/device/lib/base/mmio.h

```diff
@@ -236,6 +236,24 @@ inline void mmio_region_nonatomic_set_mask32(mmio_region_t base,
   mmio_region_write32(base, offset, value);
 }
 
+/**
+ * Sets the bits in `mask` from the MMIO region `base` at the given offset.
+ *
+ * This function is like `nonatomic_set_mask32`, but does not perform a
+ * read, for use with write-only memory.
```

Comment:
I think this comment should be expanded to say "write-only, rw1c and w1c  memory", as per @sriyerg comment, it also nicely works for those registers.

Reply:
Sure.

Reply:
Well, if I understand correcty @sriyerg experienced the issue with a `rw1c` register - `uart.INTR_STATE`. I think this function solves that use-case as well, and hence it should be used for `rw1c` and `w1c` (I guess `w1c` falls under "write-only"). It would be nice to document it for the consumers of mmio library, to make it more explicit.